docs/mgmt/mitigate_breaking_changes.md

```diff
@@ -0,0 +1,108 @@
+
+To enforce this, we have a check in `dotnet build` that will fail if a breaking change is detected. This check is performed by the APIComPat tool. The tool is run as part of the build process and will compare the current assembly with the previous assembly (if it exists). If any breaking changes are detected, the build will fail, and you will see the following error message:
+```
+error : MembersMustExist : Member '<MemberName>' does not exist in the implementation but it does exist in the contract.
+```
+
+## General Principles
+
+1. We should never introduce API breaking changes. If there are breaking changes, we should always mitigate them in a non-breaking way.
+1. We should try our best to avoid behavioral breaking changes. The way we mitigate breaking changes should not change the behavior of the API.
+1. The SDK is designed to support multiple API versions, therefore we should ensure that the mitigation steps do not introduce breaking changes for older API versions.
+
+## Mitigation Steps
+
+### `MembersMustExist` Errors
+
+`MembersMustExist` errors from ApiCompat are usually seen when a member is removed in the new version of the SDK. This can happen when a method/property/class is removed, or when a method/property/class is renamed. To mitigate this, we should introduce the removed member back and add the `[EditorBrowsable(EditorBrowsableState.Never)]` attribute to the member. This will hide the member from intellisense, and will prevent the member from being used in client code. This will not be a breaking change for clients that are using the member, but it will prevent new clients from using the member.
+
+To mitigate a `MembersMustExist` error for removed **properties** or **methods**, follow these steps:
+
+1. Create a file outside the `src/Generated` directory with the same name of its containing class. Usually they should go into the `src/Customization` or `src/Custom` directory, if there is not such a directory, please create one.
+1. Create a partial class in the file with the same name as the containing class.
+1. Introduce the property or method back (**including its original implementation and xml documentation**) with the `[EditorBrowsable(EditorBrowsableState.Never)]` attribute.
+1. Introduce everything needed in the original implementation back as well, including any helper methods, classes, etc.
+
+In addition, for removed properties, we need to do some additional steps to make sure our SDK still works for old API versions. Please see the section below for more details.
+
+If the property no longer works in all API versions, please add the `[Obsolete("<Put a message here>", false)]` attribute to the property and just leave the property with its auto getter and setter.
```

Comment:
I don't think we should just leave the auto properties. Auto properties will create a false impression that they work. We should add the property and throw NotSupportedException from both the getter and the setter.

Reply:
This deviates slightly from our old recommendations.  Original we were going to use Obsolete and eventually remove since anyone treating warnings as errors would immediately get a compilation error (better than the throw runtime error) and when we turn it to an error later everyone would get this.

Happy to change this if we think throwing on these is better but wanted to make sure.